### PR TITLE
Retain symbols of computed properties defined on classes.

### LINKED ISSUE #504: Computed property keys are mistakenly tree-shaken away
In the following code snippet, rollup drops `_x`:
```js
const _x = Symbol('x');
export class Foo {
  [_x]() {}
}
```
([see live here](http://rollupjs.org/#%7B%22options%22%3A%7B%22format%22%3A%22cjs%22%2C%22moduleName%22%3A%22myBundle%22%2C%22globals%22%3A%7B%7D%7D%2C%22modules%22%3A%5B%7B%22name%22%3A%22main.js%22%2C%22code%22%3A%22const%20_x%20%3D%20Symbol('x')%3B%5Cnconst%20_y%20%3D%20Symbol('y')%3B%5Cnexport%20class%20Foo%20%7B%5Cn%20%20%5B_x%5D()%20%7B%20return%20_y%3B%20%7D%5Cn%7D%22%7D%5D%7D))

### PR DESCRIPTION
**Pull request by ochafik**
(fixes #504, similar fix as #439)

### CODE REVIEW

File: test/form/computed-properties/main.js [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
+var bar = 'bar';
+var baz = 'baz';
+var bam = 'bam';
 
 export var x = {[foo]: 'bar'};
+
+export class X {
+  [bar]() {}
+  get [baz]() {}
+  set [bam](value) {}
```
Comment (Victorystick) [inline test/form/computed-properties/main.js]: Code in the Rollup repo should be indented using tabs where possible. Good to merge once that's fixed.

File: src/ast/isReference.js [hunk 2]
```diff
@@ -10,7 +10,10 @@ export default function isReference ( node, parent ) {
 		if ( !parent ) return true;
 
 		// TODO is this right?
-		if ( parent.type === 'MemberExpression' ) return parent.computed || node === parent.object;
+		if ( parent.type === 'MemberExpression' ||
+			parent.type === 'MethodDefinition' ) {
```
Comment (Victorystick) [inline src/ast/isReference.js]: Keep 'em on the same line.

**Comment by Rich-Harris** [comment]
Thans @ochafik! @Victorystick I applied those changes locally

**Comment by ochafik** [comment]
Woohoo, thanks guys!

**Pull request merged**